src/coreComponents/constitutive/solid/CompressibleSolid.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file CompressibleSolid.cpp
+ */
+
+#include "CompressibleSolid.hpp"
+#include "porosity/PressurePorosity.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+namespace constitutive
+{
+
+template< typename PORO_TYPE >
+CompressibleSolid< PORO_TYPE >::CompressibleSolid( string const & name, Group * const parent ):
+  CoupledSolid< NullModel, PORO_TYPE >( name, parent )
+{}
+
+template< typename PORO_TYPE >
+CompressibleSolid< PORO_TYPE >::~CompressibleSolid()
+{}
+
+// Register all CoupleSolid model types.
+typedef CompressibleSolid< PressurePorosity > CompressibleRock;
+
+REGISTER_CATALOG_ENTRY( ConstitutiveBase, CompressibleRock, string const &, Group * const )
```

Comment:
yes, I just followed what was done in other files in the code. If you think it's better without the `typedef` I can certainly remove it both in this file and in the other.

Reply:
It won't work without a typedef (although I'd prefer `using`), because the derived type name is used by `REGISTER_CATALOG_ENTRY` macro as part of the name of static variable it creates.